### PR TITLE
fix: some typing issues

### PR DESCRIPTION
**Pull request by qexat**

### CODE REVIEW

File: src/samarium/core.py [hunk 1]
```diff
@@ -97,16 +98,16 @@ def import_module(data: str, reg: Registry) -> None:
                 )
 
 
-def print_safe(*args) -> Any:
-    args = list(map(correct_type, args))
-    return_args = args[:]
-    args = list(map(str, args))
-    types = [*map(type, args)]
+def print_safe(*args: Attrs | bool | None) -> Any:
```
Comment (trag1c) [inline src/samarium/core.py:101]: Pretty sure it always returns `Attrs` (considering `bool` and `None` get converted to a Samarium type)

File: src/samarium/tokenizer.py [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
 from . import handlers
 from .exceptions import SamariumSyntaxError, handle_exception
 from .tokens import Token
 
-Tokenlike = Union[Token, str, int]
+
+if sys.version_info >= (3, 10):
+    Tokenlike = Token | str | int
+else:
+    Tokenlike = Union[Token, str, int]
```
Comment (trag1c) [inline src/samarium/tokenizer.py:14]: why tho

File: src/samarium/exceptions.py [hunk 3]
```diff
@@ -1,5 +1,5 @@
-from os import _exit
 import sys
+from os import _exit
```
Comment (trag1c) [inline src/samarium/exceptions.py:2]: Unused import

**Review by trag1c — CHANGES_REQUESTED** [review]

File: src/samarium/classes/base.py [hunk 4]
```diff
@@ -798,7 +802,7 @@ def correct_type(obj: Any) -> Any:
     return obj
 
 
-def mkslice(start: Any = None, stop: Any = None, step: Any = None) -> Any:
+def mkslice(start: Any = None, stop: Any = None, step: Any = None) -> Slice:
```
[the remark this answers is not in the document]
Comment (qexat) [reply src/samarium/classes/base.py:805]:
```suggestion
def mkslice(start: Any = None, stop: Any = None, step: Any = None) -> Any:
```

File: src/samarium/core.py (hunk 1 above)
Comment (qexat) [reply src/samarium/core.py:101]:
```suggestion
def print_safe(*args: Attrs | bool | None) -> Attrs:
```

File: src/samarium/tokenizer.py (hunk 2 above)
Comment (qexat) [reply src/samarium/tokenizer.py:14]: idk

File: src/samarium/exceptions.py (hunk 3 above)
Comment (qexat) [reply src/samarium/exceptions.py:2]:
```suggestion
```

File: src/samarium/tokenizer.py (hunk 2 above)
Comment (qexat) [reply src/samarium/tokenizer.py:14]:
```suggestion
Tokenlike = Union[Token, str, int]
```

File: src/samarium/tokenizer.py [hunk 5]
```diff
@@ -1,11 +1,13 @@
 from contextlib import suppress
 from string import ascii_letters, digits
+import sys
 from typing import Union
```
Comment (trag1c) [inline src/samarium/tokenizer.py:4]: Unused import gg
```suggestion
from typing import Union
```

**Review by trag1c — CHANGES_REQUESTED** [review]

File: src/samarium/handlers.py [hunk 6]
```diff
@@ -11,7 +11,7 @@ def pointer(self) -> str:
         return self.program[:1]
 
     def next(self, offset: int = 1) -> str:
-        return self.program[offset:offset + 1]
+        return self.program[offset : offset + 1]
```
Comment (qexat) [inline src/samarium/handlers.py:14]: black is drunk :skull:

File: src/samarium/template.py [hunk 7]
```diff
@@ -4,7 +4,7 @@
 
 STDOUT = sys.stdout
 sys.stdout = open(os.devnull, "w")
-{{ CODE }}
+{{CODE}}
```
Comment (qexat) [inline src/samarium/template.py:7]: :skull:

File: src/samarium/utils.py [hunk 8]
```diff
@@ -146,7 +150,7 @@ def smformat(string: str, fields: str | list[Any] | dict[Any, Any]) -> str:
     if isinstance(fields, dict):
         it = fields.items()
     for k, v in it:
-        string = sub(fr"(?<!\$)\${k}", str(v), string)
+        string = sub(rf"(?<!\$)\${k}", str(v), string)
```
Comment (qexat) [inline src/samarium/utils.py:153]: :fr: :x:

File: src/samarium/handlers.py (hunk 6 above)
Comment (qexat) [reply src/samarium/handlers.py:14]:
```suggestion
        return self.program[offset:offset + 1]
```

File: src/samarium/template.py (hunk 7 above)
Comment (qexat) [reply src/samarium/template.py:7]:
```suggestion
{{ CODE }}
```

File: src/samarium/classes/base.py [hunk 9]
```diff
@@ -245,7 +252,9 @@ def __init__(self, v: Any = None) -> None:
         elif isinstance(v, String):
             self.val = parse_integer(v.val)
         else:
-            raise SamariumTypeError(f"cannot cast {get_type_name(v)} to Integer")
+            raise SamariumTypeError(
+                f"cannot cast {get_type_name(v)} to Integer"
+            )
```
Comment (qexat) [inline src/samarium/classes/base.py:257]: needs revert

File: src/samarium/classes/base.py [hunk 10]
```diff
@@ -845,22 +862,24 @@ def modify(
...
                 errmsg = str(e)
                 if "positional argument: 'self'" in errmsg:
                     raise SamariumTypeError("missing instance")
                 missing_args = MISSING_ARGS_PATTERN.search(errmsg)
                 if missing_args:
                     given = argc - (int(missing_args.group(1)) or 1)
-                    raise SamariumTypeError(f"not enough arguments ({given}/{argc})")
+                    raise SamariumTypeError(
+                        f"not enough arguments ({given}/{argc})"
+                    )
```
Comment (qexat) [inline src/samarium/classes/base.py:882]: needs revert

File: src/samarium/classes/fileio.py [hunk 11]
```diff
@@ -88,7 +90,9 @@ def quick(
 class File(Attrs):
     __slots__ = ("binary", "mode", "path", "val")
 
-    def __init__(self, file: IO, mode: str, path: str | int, binary: bool) -> None:
+    def __init__(
+        self, file: IO, mode: str, path: str | int, binary: bool
+    ) -> None:
```
Comment (qexat) [inline src/samarium/classes/fileio.py:95]: needs revert

File: src/samarium/classes/misc.py [hunk 12]
```diff
@@ -3,7 +3,11 @@
 from typing import Any, Iterable
 from typing import Iterator as Iter
 
-from ..exceptions import SamariumSyntaxError, SamariumTypeError, SamariumValueError
+from ..exceptions import (
+    SamariumSyntaxError,
+    SamariumTypeError,
+    SamariumValueError,
+)
```
Comment (qexat) [inline src/samarium/classes/misc.py:10]: needs revert

File: src/samarium/classes/misc.py [hunk 13]
```diff
@@ -69,7 +73,9 @@ class Iterator(Attrs):
 
     def __init__(self, value: Any) -> None:
         if not isinstance(value, Iterable):
-            raise SamariumTypeError("cannot create an Iterator from a non-iterable")
+            raise SamariumTypeError(
+                "cannot create an Iterator from a non-iterable"
+            )
```
Comment (qexat) [inline src/samarium/classes/misc.py:78]: needs revert

File: src/samarium/core.py [hunk 14]
```diff
@@ -130,8 +132,10 @@ def run(
     Runtime.quit_on_error = quit_on_error
     code = Transpiler(tokenize(code), reg).transpile().output
     if load_template:
-        code = (Path(__file__).resolve().parent / "template.py").read_text().replace(
-            "{{ CODE }}", code
+        code = (
+            (Path(__file__).resolve().parent / "template.py")
+            .read_text()
+            .replace("{{ CODE }}", code)
```
Comment (qexat) [inline src/samarium/core.py:138]: needs revert

File: src/samarium/shell.py [hunk 15]
```diff
@@ -55,7 +55,8 @@ def execute(self, code: str) -> None:
 
     def render(self, window: FullscreenWindow) -> None:
         window.render_to_terminal(
-            self.messages, (len(self.messages) - 1, len(clean_ansi(self.messages[-1])))
+            self.messages,
+            (len(self.messages) - 1, len(clean_ansi(self.messages[-1]))),
```
Comment (qexat) [inline src/samarium/shell.py:59]: needs revert (wtf??)

File: src/samarium/shell.py [hunk 16]
```diff
@@ -83,14 +84,18 @@ def handle(self, e: Event | str | None) -> bool | None:
         elif e == "<BACKSPACE>":
             curr = self.messages[-1]
             if len(curr) + 4:
-                self.messages[-1] = self.prompt + curr[len(self.prompt) : len(curr) - 1]
+                self.messages[-1] = (
+                    self.prompt + curr[len(self.prompt) : len(curr) - 1]
+                )
```
Comment (qexat) [inline src/samarium/shell.py:89]: needs revert

File: src/samarium/shell.py [hunk 17]
```diff
@@ -83,14 +84,18 @@ def handle(self, e: Event | str | None) -> bool | None:
...
         elif e == "<Ctrl-j>":
             if self.on_return() is False:
                 return None
         elif isinstance(e, PasteEvent):
             self.on_paste(e)
         elif (
-            isinstance(e, Event) or e is None or (e.startswith("<") and e.endswith(">"))
+            isinstance(e, Event)
+            or e is None
+            or (e.startswith("<") and e.endswith(">"))
```
Comment (qexat) [inline src/samarium/shell.py:98]: needs revert

File: src/samarium/template.py [hunk 18]
```diff
@@ -16,6 +16,8 @@
     elif argc.val == 1:
         ex = entry(Array(map(String, sys.argv[1:])))
     else:
-        raise SamariumSyntaxError("entry function should take 0 or 1 arguments")
+        raise SamariumSyntaxError(
+            "entry function should take 0 or 1 arguments"
+        )
```
Comment (qexat) [inline src/samarium/template.py:21]: needs revert

File: src/samarium/tokens.py [hunk 19]
```diff
@@ -114,7 +114,9 @@ class Token(Enum):
 
 
 FILE_IO_TOKENS = [
-    token for name, token in Token.__members__.items() if name.startswith("FILE_")
+    token
+    for name, token in Token.__members__.items()
+    if name.startswith("FILE_")
```
Comment (qexat) [inline src/samarium/tokens.py:119]: needs revert

File: src/samarium/transpiler.py [hunk 20]
```diff
@@ -346,7 +346,9 @@ def _file_io(self) -> None:
 
         self._file_token = None
 
-        open_template = f"{before_token}=FileManager.{{}}({after_token}, Mode.{{}})"
+        open_template = (
+            f"{before_token}=FileManager.{{}}({after_token}, Mode.{{}})"
+        )
```
Comment (qexat) [inline src/samarium/transpiler.py:351]: needs revert

File: src/samarium/transpiler.py [hunk 21]
```diff
@@ -437,14 +439,19 @@ def _brackets(self, token: Token) -> None:
         elif token is Token.BRACE_CLOSE:
             self._indent -= 1
             self._reg[Switch.FUNCTION] = False
-            if self._reg[Switch.CLASS] and self._indent == self._class_indent[-1]:
+            if (
+                self._reg[Switch.CLASS]
+                and self._indent == self._class_indent[-1]
+            ):
```
Comment (qexat) [inline src/samarium/transpiler.py:445]: needs revert

File: src/samarium/transpiler.py [hunk 22]
```diff
@@ -437,14 +439,19 @@ def _brackets(self, token: Token) -> None:
...
                 self._reg[Switch.CLASS] = False
                 self._class_indent.pop()
 
             if self._scope.current == "enum":
                 self._line.append('""")')
 
-            if (self._processed_tokens or self._line_tokens)[-1] is Token.BRACE_OPEN:
+            if (self._processed_tokens or self._line_tokens)[
+                -1
+            ] is Token.BRACE_OPEN:
```
Comment (qexat) [inline src/samarium/transpiler.py:454]: needs revert

File: src/samarium/transpiler.py [hunk 23]
```diff
@@ -514,7 +521,8 @@ def _functions(self, token: Token) -> None:
                         (
                             # Adding the self parameter for class methods
                             ["self"]
-                            if self._reg[Switch.CLASS] and self._scope.parent == "class"
+                            if self._reg[Switch.CLASS]
+                            and self._scope.parent == "class"
```
Comment (qexat) [inline src/samarium/transpiler.py:525]: needs revert

File: src/samarium/transpiler.py [hunk 24]
```diff
@@ -524,7 +532,9 @@ def _functions(self, token: Token) -> None:
             ]
         elif token is Token.DEFAULT:
             self._line.append(
-                " = {0} if {0} is not MISSING else ".format("".join(self._line).strip())
+                " = {0} if {0} is not MISSING else ".format(
+                    "".join(self._line).strip()
+                )
```
Comment (qexat) [inline src/samarium/transpiler.py:537]: needs revert

File: src/samarium/transpiler.py [hunk 25]
```diff
@@ -539,11 +549,15 @@ def _functions(self, token: Token) -> None:
     def _multisemantic(self, token: Token) -> None:
         index = self._index
         if token is Token.TRY:
-            self._line.append("try" if is_first_token(self._line) else ".random")
+            self._line.append(
+                "try" if is_first_token(self._line) else ".random"
+            )
```
Comment (qexat) [inline src/samarium/transpiler.py:554]: needs revert

File: src/samarium/transpiler.py [hunk 26]
```diff
@@ -539,11 +549,15 @@ def _functions(self, token: Token) -> None:
...
         elif token is Token.TO:
             if self._tokens[index - 1] is Token.TABLE_OPEN:
                 self._line.append("NULL")
-            self._line.append("continue" if is_first_token(self._line) else ":")
+            self._line.append(
+                "continue" if is_first_token(self._line) else ":"
+            )
```
Comment (qexat) [inline src/samarium/transpiler.py:560]: needs revert

File: src/samarium/transpiler.py [hunk 27]
```diff
@@ -570,7 +584,11 @@ def _multisemantic(self, token: Token) -> None:
                 self._line.append("class ")
             else:
                 indented = self._indent > 0
-                self._line = [*self._line[:indented], "@", *self._line[indented:]]
+                self._line = [
+                    *self._line[:indented],
+                    "@",
+                    *self._line[indented:],
+                ]
```
Comment (qexat) [inline src/samarium/transpiler.py:591]: needs revert

File: src/samarium/transpiler.py [hunk 28]
```diff
@@ -587,7 +605,10 @@ def _control_flow(self, token: Token) -> None:
             self._line.append(shift + "else ")
         else:  # FOR
             index = self._index
-            if self._scope.current == "slice" and self._tokens[index + 1] is Token.ATTR:
+            if (
+                self._scope.current == "slice"
+                and self._tokens[index + 1] is Token.ATTR
+            ):
```
Comment (qexat) [inline src/samarium/transpiler.py:611]: needs revert

File: src/samarium/transpiler.py [hunk 29]
```diff
@@ -609,9 +630,14 @@ def _core(self, token: Token) -> None:
                     self._line.append("Integer(0)")
                 self._line.append(")")
                 self._reg[Switch.BUILTIN] = False
-            if all(i in self._line_tokens for i in (Token.ASSIGN, Token.SLICE_CLOSE)):
+            if all(
+                i in self._line_tokens
+                for i in (Token.ASSIGN, Token.SLICE_CLOSE)
+            ):
```
Comment (qexat) [inline src/samarium/transpiler.py:636]: needs revert

File: src/samarium/transpiler.py [hunk 30]
```diff
@@ -609,9 +630,14 @@ def _core(self, token: Token) -> None:
...
                 if (
-                    self._line_tokens[self._line_tokens.index(Token.ASSIGN) - 1]
+                    self._line_tokens[
+                        self._line_tokens.index(Token.ASSIGN) - 1
+                    ]
```
Comment (qexat) [inline src/samarium/transpiler.py:640]: needs revert

File: src/samarium/transpiler.py [hunk 31]
```diff
@@ -629,7 +655,10 @@ def _core(self, token: Token) -> None:
                 self._line.append(f";{variable}=correct_type({variable})")
             self._submit_line()
         elif token is Token.ASSIGN:
-            if self._line_tokens.count(token) > 1 and self._scope.current != "enum":
+            if (
+                self._line_tokens.count(token) > 1
+                and self._scope.current != "enum"
+            ):
```
Comment (qexat) [inline src/samarium/transpiler.py:661]: needs revert

File: src/samarium/transpiler.py [hunk 32]
```diff
@@ -643,26 +672,34 @@ def _core(self, token: Token) -> None:
             self._line.append(".")
         elif token is Token.INSTANCE:
             if not self._reg[Switch.CLASS]:
-                throw_syntax("instance operator cannot be used outside a class")
+                throw_syntax(
+                    "instance operator cannot be used outside a class"
+                )
```
Comment (qexat) [inline src/samarium/transpiler.py:677]: needs revert

File: src/samarium/transpiler.py [hunk 33]
```diff
@@ -643,26 +672,34 @@ def _core(self, token: Token) -> None:
...
             self._line.append("self")
         elif token is Token.SLICE_OPEN:
             self._scope.enter("slice")
             assign_index = find_next(self._tokens, Token.ASSIGN, after=index)
             end_index = find_next(self._tokens, Token.END, after=index)
-            brace_index = find_next(self._tokens, Token.BRACE_OPEN, after=index)
+            brace_index = find_next(
+                self._tokens, Token.BRACE_OPEN, after=index
+            )
```
Comment (qexat) [inline src/samarium/transpiler.py:685]: needs revert

File: src/samarium/transpiler.py [hunk 34]
```diff
@@ -643,26 +672,34 @@ def _core(self, token: Token) -> None:
...
             self._slice_assign = (
                 0
                 < assign_index
                 < min(filter(lambda x: x >= 0, (brace_index, end_index)))
             )
-            self._slice_object = self._tokens[index - 1] in SLICE_OBJECT_TRIGGERS
+            self._slice_object = (
+                self._tokens[index - 1] in SLICE_OBJECT_TRIGGERS
+            )
```
Comment (qexat) [inline src/samarium/transpiler.py:693]: needs revert

File: src/samarium/transpiler.py [hunk 35]
```diff
@@ -643,26 +672,34 @@ def _core(self, token: Token) -> None:
...
             if not self._slice_object:
                 self._line.append(f".__{'gs'[self._slice_assign]}etitem__(")
             self._line.append("mkslice(t(")
         elif token is Token.SLICE_CLOSE:
             if not self._slice_assign:
                 self._scope.exit()
-            self._line.append(")" * (3 - self._slice_object - self._slice_assign))
+            self._line.append(
+                ")" * (3 - self._slice_object - self._slice_assign)
+            )
```
Comment (qexat) [inline src/samarium/transpiler.py:702]: needs revert

File: src/samarium/transpiler.py [hunk 36]
```diff
@@ -693,7 +730,12 @@ def _builtins(self, token: Token) -> None:
             if self._line_tokens[-2] in Group.operators:
                 self._line.append("NULL")
             indented = self._indent > 0
-            self._line = [*self._line[:indented], "throw(", *self._line[indented:], ")"]
+            self._line = [
+                *self._line[:indented],
+                "throw(",
+                *self._line[indented:],
+                ")",
+            ]
```
Comment (qexat) [inline src/samarium/transpiler.py:738]: needs revert

File: src/samarium/transpiler.py [hunk 37]
```diff
@@ -709,7 +751,12 @@ def _builtins(self, token: Token) -> None:
                 hook = self._line.index("=") + 1
             else:
                 hook = self._indent > 0
-            self._line = [*self._line[:hook], "print_safe(", *self._line[hook:], ")"]
+            self._line = [
+                *self._line[:hook],
+                "print_safe(",
+                *self._line[hook:],
+                ")",
+            ]
```
Comment (qexat) [inline src/samarium/transpiler.py:759]: needs revert

File: src/samarium/utils.py [hunk 38]
```diff
@@ -64,16 +65,18 @@ def __delitem__(self, item: KT) -> None:
...
-    _instances = {}
+    _instances: dict[type[Singleton], Singleton] = {}
 
-    def __new__(cls, *args, **kwargs):
+    def __new__(cls, *args: Any, **kwargs: Any):
         if cls not in cls._instances:
-            cls._instances[cls] = super(Singleton, cls).__new__(cls, *args, **kwargs)
+            cls._instances[cls] = super(Singleton, cls).__new__(
+                cls, *args, **kwargs
+            )
```
Comment (qexat) [inline src/samarium/utils.py:74]: needs revert

File: src/samarium/utils.py [hunk 39]
```diff
@@ -93,7 +96,9 @@ def match_brackets(tokens_: list[Tokenlike]) -> tuple[int, list[Token]]:
     return 0, []
 
 
-def run_with_backup(main: Callable[..., T], backup: Callable[..., T], *args) -> T:
+def run_with_backup(
+    main: Callable[..., T], backup: Callable[..., T], *args: Any
+) -> T:
```
Comment (qexat) [inline src/samarium/utils.py:101]: needs revert

File: src/samarium/classes/base.py [hunk 40]
```diff
@@ -129,7 +134,9 @@ def hash(self) -> Integer:
         v = hsh()
         if isinstance(v, Integer):
             return v
-        raise SamariumTypeError(f"{get_type_name(self)}## returned a non-integer")
+        raise SamariumTypeError(
+            f"{get_type_name(self)}## returned a non-integer"
+        )
```
[the remark this answers is not in the document]
Comment (qexat) [reply src/samarium/classes/base.py:137]:
```suggestion
        raise SamariumTypeError(f"{get_type_name(self)}## returned a non-integer")
```

File: src/samarium/classes/base.py (hunk 9 above)
Comment (qexat) [reply src/samarium/classes/base.py:253]:
```suggestion
            raise SamariumTypeError(f"cannot cast {get_type_name(v)} to Integer")
```

File: src/samarium/classes/base.py [hunk 41]
```diff
@@ -466,12 +475,16 @@ def __init__(self, value: Any = None) -> None:
...
+                    "cannot convert an infinite Slice to Array"
+                )
             self.val = list(value.range)
         elif isinstance(value, Iterable):
             self.val = [*value]
         else:
-            raise SamariumTypeError(f"cannot cast {get_type_name(value)} to Array")
+            raise SamariumTypeError(
+                f"cannot cast {get_type_name(value)} to Array"
+            )
```
[the remark this answers is not in the document]
Comment (qexat) [reply src/samarium/classes/base.py:479]:
```suggestion
            raise SamariumTypeError(f"cannot cast {get_type_name(value)} to Array")
```

File: src/samarium/classes/base.py [hunk 42]
```diff
@@ -581,7 +594,9 @@ def __init__(self, value: Any = None) -> None:
             self.val = value
         elif isinstance(value, Array):
             arr = value.val
-            if all(isinstance(i, (String, Array)) and len(i.val) == 2 for i in arr):
+            if all(
+                isinstance(i, (String, Array)) and len(i.val) == 2 for i in arr
+            ):
```
[the remark this answers is not in the document]
Comment (qexat) [reply src/samarium/classes/base.py:589]:
```suggestion
            if all(isinstance(i, (String, Array)) and len(i.val) == 2 for i in arr):
```

File: src/samarium/classes/base.py [hunk 43]
```diff
@@ -597,7 +612,9 @@ def __init__(self, value: Any = None) -> None:
                     "not all elements of the array are of length 2"
                 )
         else:
-            raise SamariumTypeError(f"cannot cast {get_type_name(value)} to Table")
+            raise SamariumTypeError(
+                f"cannot cast {get_type_name(value)} to Table"
+            )
```
[the remark this answers is not in the document]
Comment (qexat) [reply src/samarium/classes/base.py:605]:
```suggestion
            raise SamariumTypeError(f"cannot cast {get_type_name(value)} to Table")
```

File: src/samarium/classes/fileio.py [hunk 44]
```diff
@@ -24,7 +24,9 @@ def create(path: String) -> Null:
         return NULL
 
     @staticmethod
-    def open(path: String | Integer, mode: Mode, *, binary: bool = False) -> File:
+    def open(
+        path: String | Integer, mode: Mode, *, binary: bool = False
+    ) -> File:
```
[the remark this answers is not in the document]
Comment (qexat) [reply src/samarium/classes/fileio.py:29]:
```suggestion
    def open(path: String | Integer, mode: Mode, *, binary: bool = False) -> File:
```

File: src/samarium/classes/misc.py (hunk 12 above)
Comment (qexat) [reply src/samarium/classes/misc.py:10]:
```suggestion
from ..exceptions import SamariumSyntaxError, SamariumTypeError, SamariumValueError
```

File: src/samarium/transpiler.py (hunk 36 above)
Comment (qexat) [reply src/samarium/transpiler.py:705]:
```suggestion
            self._line = [*self._line[:indented], "throw(", *self._line[indented:], ")"]
```

File: src/samarium/core.py (hunk 14 above)
Comment (qexat) [reply src/samarium/core.py:136]:
```suggestion
        code = ((Path(__file__).resolve().parent / "template.py").read_text().replace("{{ CODE }}", code)
```

Comment (qexat) [reply src/samarium/core.py:136]: @trag1c does this one count

File: src/samarium/shell.py (hunk 15 above)
Comment (qexat) [reply src/samarium/shell.py:59]:
```suggestion
            self.messages, (len(self.messages) - 1, len(clean_ansi(self.messages[-1]))),
```

File: src/samarium/transpiler.py (hunk 37 above)
Comment (qexat) [reply src/samarium/transpiler.py:721]:
```suggestion
            self._line = [*self._line[:hook], "print_safe(", *self._line[hook:], ")"]
```

File: src/samarium/transpiler.py (hunk 27 above)
Comment (qexat) [reply src/samarium/transpiler.py:577]:
```suggestion
                self._line = [*self._line[:indented], "@", *self._line[indented:]]
```

File: src/samarium/core.py [hunk 45]
```diff
@@ -130,8 +130,8 @@ def run(
     Runtime.quit_on_error = quit_on_error
     code = Transpiler(tokenize(code), reg).transpile().output
     if load_template:
-        code = (Path(__file__).resolve().parent / "template.py").read_text().replace(
-            "{{ CODE }}", code
+        code = ((Path(__file__).resolve().parent / "template.py").read_text().replace(
+            "{{ CODE }}", code)
```
Comment (qexat) [inline src/samarium/core.py:134]: stupid ass @qexat 
```suggestion
        code = (Path(__file__).resolve().parent / "template.py").read_text().replace(
            "{{ CODE }}", code
```

File: src/samarium/shell.py [hunk 46]
```diff
@@ -55,7 +55,7 @@ def execute(self, code: str) -> None:
 
     def render(self, window: FullscreenWindow) -> None:
         window.render_to_terminal(
-            self.messages, (len(self.messages) - 1, len(clean_ansi(self.messages[-1])))
+            self.messages, (len(self.messages) - 1, len(clean_ansi(self.messages[-1]))),
```
Comment (qexat) [inline src/samarium/shell.py:58]:
```suggestion
            self.messages, (len(self.messages) - 1, len(clean_ansi(self.messages[-1])))
```

File: src/samarium/template.py (hunk 7 above)
Comment (qexat) [inline src/samarium/template.py:7]:
```suggestion
{{ CODE }}
```

**Review by qexat — COMMENTED** [review]
:skull:

**Review by trag1c — APPROVED** [review]

**Comment by trag1c** [comment]
Thanks ❤️

**Comment by qexat** [comment]
you're welcome, sorry for the commit history :sob:

**Pull request merged**